R/check_dhis2_attributes-helpers.R

```diff
@@ -0,0 +1,248 @@
+#' Get the dataset identifiers and names
+#'
+#' @param username the user name
+#' @param password the user's password
+#' @param base_url the base URL of the DHIS2 server
+#'
+#' @export
+#'
+get_data_sets <- function(base_url, username, password) {
+  checkmate::assertCharacter(base_url, len = 1L, null.ok = FALSE,
+                             any.missing = FALSE)
+  checkmate::assertCharacter(username, len = 1L, null.ok = FALSE,
+                             any.missing = FALSE)
+  checkmate::assertCharacter(password, len = 1L, null.ok = FALSE,
+                             any.missing = FALSE)
+
+  url <- paste0(base_url, "/api/dataSets?fields=id,name,shortName&paging=false")
```

Comment:
Consider passing this as the default value to the function args to avoid recompiling code in future if DHIS decides to use a different path to access the same dataset. e.g. `datasets = "/api/dataSets?fields=id,name,shortName&paging=false" `

Reply:
Good point!

In `check_dhis2_attributes_helpers.R`, I have updated/removed many functions. I believe that this architecture will not be changed by the DHIS2 people. However, if that is the case, the connexion is made within the `make_api_request()`, which will then need to be updated. The remaining functions, that rely on this one, will not need to be updated.